sdk/java/pulumi/src/main/java/com/pulumi/provider/internal/infer/ComponentAnalyzer.java

```diff
@@ -68,25 +68,33 @@ public static PackageSpec generateSchema(Metadata metadata, Class<?>... classes)
 
         ComponentAnalyzer analyzer = new ComponentAnalyzer(metadata);
         Map<String, ResourceSpec> components = new HashMap<>();
-        
+
+        if (classes.length == 0) {
+            throw new IllegalArgumentException("At least one component class must be provided");
+        }
+
+        String namespace = classes[0].getPackage().getName().split("\\.")[1];
```

Comment:
Not sure if we should error out if not all classes have the same namespace?

Reply:
Probably a safer way to go

Reply:
Done, added an exception.